### PR TITLE
ZK-4068: datebox failing date format

### PR DESCRIPTION
**Pull request by plijyfes**

### CODE REVIEW

File: zul/src/archive/web/js/zul/db/datefmt.js [hunk 1]
```diff
@@ -221,7 +221,7 @@ zk.fmt.Date = {
 				var nosep, nv; //no separator
 				if (k < fl) {
 					var c2 = fmt.charAt(k);
-					nosep = c2 == 'y' || c2 == 'M' || c2 == 'd' || c2 == 'E' || c2 == 'm' || c2 == 's' || c2 == 'S';
+					nosep = c2 == 'y' || c2 == 'M' || c2 == 'd' || c2 == 'E' || c2 == 'm' || c2 == 's' || c2 == 'S' || c2 == 'h' || c2 == 'H' || c2 == 'k' || c2 == 'K' || c2 == 'a' || c2 == 'A';
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/db/datefmt.js:224]: 'yMd....'.indexOf(c2) != -1

File: zul/src/archive/web/js/zul/db/datefmt.js [hunk 2]
```diff
@@ -262,7 +262,7 @@ zk.fmt.Date = {
 					break;
 				case 'M':
 					var mon = token ? token.toLowerCase() : '',
-						isNumber0 = !isNaN(token);
+						isNumber0 = !isNaN(token) || len <= 2;
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/db/datefmt.js:265]: comment why

**Review by scribetw — APPROVED** [review]

**Review by scribetw — CHANGES_REQUESTED** [review]
refine

**Review by scribetw — APPROVED** [review]

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**